src/main/kotlin/org/rust/ide/formatter/processors/RsTrailingCommaFormatProcessor.kt

```diff
@@ -97,10 +96,11 @@ fun CommaList.addTrailingCommaForElement(list: PsiElement): Boolean {
     return true
 }
 
-fun CommaList.isLastElement(list: PsiElement, element: PsiElement): Boolean {
+fun CommaList.isOnSameLineAsLastElement(list: PsiElement, element: PsiElement, document: Document): Boolean {
     check(list.elementType == this.list && isElement(element))
     val rbrace = list.lastChild
     if (rbrace.elementType != closingBrace) return false
-    val lastElement = rbrace.getPrevNonCommentSibling() ?: return false
-    return lastElement == element
+    val lastElement = rbrace.getPrevNonCommentSibling()?.takeIf(isElement) ?: return false
+    return lastElement == element ||
+        document.getLineNumber(lastElement.endOffset) == document.getLineNumber(element.startOffset)
```

Comment:
```suggestion
        document.getLineNumber(lastElement.endOffset) == document.getLineNumber(element.endOffset)
```
Not sure but it looks more reasonable to use `endOffset`. For example, it will work when `element` is multi-line (e.g. a struct initialization) and the last element is located on the same line. WDYT?

Reply:
Fixed it and added a [test](https://github.com/t-kameyama/intellij-rust/blob/25121a99efafc2d9a9171d1b3c0af73d5d64f706/src/test/kotlin/org/rust/ide/actions/mover/RsCommaListElementUpDownMoverTest.kt#L372-L392).